src/foraging_gui/Dialogs.py

```diff
@@ -206,43 +206,31 @@ def _activated_4(self):
-                    ItemsLeft.append(str(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Left']['LaserPowerVoltage'][i]))
-                for i in range(len(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Right']['LaserPowerVoltage'])):
-                    ItemsRight.append(str(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Right']['LaserPowerVoltage'][i]))
-
-            elif Protocol=='Constant' or Protocol=='Pulse':
-                for i in range(len(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Left']['LaserPowerVoltage'])):
-                    ItemsLeft.append(str(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Left']['LaserPowerVoltage'][i]))
-                for i in range(len(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Right']['LaserPowerVoltage'])):
-                    ItemsRight.append(str(RecentLaserCalibration[Color][Protocol][CurrentFrequency]['Right']['LaserPowerVoltage'][i]))
-            ItemsLeft=sorted(ItemsLeft)
-            ItemsRight=sorted(ItemsRight)
-            eval('self.LaserPowerLeft_'+str(Numb)+'.clear()')
-            eval('self.LaserPowerLeft_'+str(Numb)+'.addItems(ItemsLeft)')
-            if eval('self.LaserPowerLeft_'+str(Numb)+'.findText(CurrentlaserPowerLeft)'):
-                index = eval('self.LaserPowerLeft_'+str(Numb)+'.findText(CurrentlaserPowerLeft)')
-                if index != -1:
-                    eval('self.LaserPowerLeft_'+str(Numb)+'.setCurrentIndex(index)')
-            eval('self.LaserPowerRight_'+str(Numb)+'.clear()')
-            eval('self.LaserPowerRight_'+str(Numb)+'.addItems(ItemsRight)')
-            if eval('self.LaserPowerRight_'+str(Numb)+'.findText(CurrentlaserPowerRight)'):
-                index = eval('self.LaserPowerRight_'+str(Numb)+'.findText(CurrentlaserPowerRight)')
-                if index != -1:
-                    eval('self.LaserPowerRight_'+str(Numb)+'.setCurrentIndex(index)')
+            for laser_tag in laser_tags:
+                ItemsLaserPower=[]
+                CurrentlaserPowerLaser=eval(f"self.Laser{str(laser_tag)}_power_{str(Numb)}.currentText()")
+                if Protocol in ['Sine']:
+                    for i in range(len(RecentLaserCalibration[Color][Protocol][CurrentFrequency][f"Laser_{str(laser_tag)}"]['LaserPowerVoltage'])):
+                        ItemsLaserPower.append(str(RecentLaserCalibration[Color][Protocol][CurrentFrequency][f"Laser_{str(laser_tag)}"]['LaserPowerVoltage'][i]))
+                if Protocol in ['Constant','Pulse']:
+                    for i in range(len(RecentLaserCalibration[Color][Protocol][f"Laser_{str(laser_tag)}"]['LaserPowerVoltage'])):
+                        ItemsLaserPower.append(str(RecentLaserCalibration[Color][Protocol][f"Laser_{str(laser_tag)}"]['LaserPowerVoltage'][i]))
+                ItemsLaserPower=sorted(ItemsLaserPower)
+                eval(f"self.Laser{str(laser_tag)}_power_{str(Numb)}.clear()")
```

Comment:
```suggestion
                getattr(self, f"Laser{str(laser_tag)}_power_{str(Numb)}").clear()
```
for all instances like this.

Reply:
Updated all of them